sdk/tests/utilities/id_generator.py

```diff
@@ -47,6 +47,31 @@ def generate_scope_and_code(self, entity, scope=None, code_prefix=None):
         self._scope_and_codes.add(item)
         return item
 
+    def add_scope_and_code(self, entity, scope, code):
```

Comment:
Given the user was explicitly adding in externally generated values, I thought it would better to remove ambiguity and force them to add all the values.

Reply:
ok - that makes sense